src/modules/account/components/account-nav/index.tsx

```diff
@@ -1,167 +1,114 @@
-"use client"
+'use client';
 
-import { Customer } from "@medusajs/medusa"
-import { clx } from "@medusajs/ui"
-import { ArrowRightOnRectangle } from "@medusajs/icons"
-import { useParams, usePathname } from "next/navigation"
+import {
+  CubeIcon,
+  UserCircleIcon,
+  TruckIcon,
+  CreditCardIcon,
+  ArrowUpTrayIcon,
+} from '@heroicons/react/24/outline';
+import { useDictionary } from '@lib/context/dictionary-context';
+import { conditionalClassNames } from '@lib/util/conditional-classname';
+import { Customer } from '@medusajs/medusa';
+import { signOut } from '@modules/account/actions';
+import IconButton from '@modules/common/components/icon-button';
+import { usePathname, useRouter } from 'next/navigation';
+import { useMemo } from 'react';
+import AccountAvatar from '../account-avatar';
 
-import ChevronDown from "@modules/common/icons/chevron-down"
-import { signOut } from "@modules/account/actions"
-import User from "@modules/common/icons/user"
-import MapPin from "@modules/common/icons/map-pin"
-import Package from "@modules/common/icons/package"
-import LocalizedClientLink from "@modules/common/components/localized-client-link"
+const getSidebarItems = (
+  dictionary: Record<string, any>,
+  options: { orderCount: number }
+) => {
+  const sideBar = dictionary.account.sideBar;
+
+  return [
+    {
+      title: sideBar.order,
+      href: 'order?status=all',
+      icon: <CubeIcon className="w-6 h-6" />,
+      count: options.orderCount,
+    },
+    {
+      title: sideBar.profile,
+      href: 'profile',
+      icon: <UserCircleIcon className="w-6 h-6" />,
+    },
+    {
+      title: sideBar.delivery,
+      href: 'delivery',
+      icon: <TruckIcon className="w-6 h-6" />,
+    },
+    {
+      title: sideBar.payment,
+      href: 'payment',
+      icon: <CreditCardIcon className="w-6 h-6" />,
+    },
+  ];
+};
 
 const AccountNav = ({
   customer,
+  orderCount,
 }: {
-  customer: Omit<Customer, "password_hash"> | null
+  customer: Omit<Customer, 'password_hash'> | null;
+  orderCount?: number;
 }) => {
-  const route = usePathname()
-  const { countryCode } = useParams()
+  const pathName = usePathname();
+  const router = useRouter();
+  const dictionary = useDictionary();
+  const sideBarItems = getSidebarItems(dictionary, { orderCount });
+
+  const splittedPath = useMemo(() => pathName.split('/'), [pathName]);
```

Comment:
Using split on `pathName` is not recommended. It might be helpful to use radio buttons for the `sideBarItems`

Reply:
I cannot think of any other way to get the current selection without getting it from the pathname. Otherwise it will be behaved unexpectedly